File: app/src/main/java/com/softwareverde/tidyduck/most/PrimitiveType.java
package com.softwareverde.tidyduck.most;

public class PrimitiveType {
    private long _id;
    private String _name;
    private boolean _isPreloadedType;
    private boolean _isNumberBaseType;
    private boolean _isStreamParameterType;
    private boolean _isArrayType;
    private boolean _isRecordType;

    public long getId() {
        return _id;
    }

    public void setId(long id) {
        _id = id;
    }

    public String getName() {
        return _name;
    }

    public void setName(String name) {
        _name = name;
    }

    public boolean isPreloadedType() {
        return _isPreloadedType;
    }

    public void setIsPreloadedType(boolean preloadedType) {
        _isPreloadedType = preloadedType;
    }

    public boolean isNumberBaseType() {
        return _isNumberBaseType;
    }

    public void setIsNumberBaseType(boolean numberBaseType) {
        _isNumberBaseType = numberBaseType;
    }

    public boolean isStreamParameterType() {
        return _isStreamParameterType;
    }

    public void setIsStreamParameterType(boolean streamParameterType) {
        _isStreamParameterType = streamParameterType;
    }

    public boolean isArrayType() {
        return _isArrayType;
    }

    public void setIsArrayType(boolean arrayType) {
        _isArrayType = arrayType;
    }

    public boolean isRecordType() {
        return _isRecordType;
    }

    public void setIsRecordType(boolean recordType) {
        _isRecordType = recordType;
    }
}

File: app/src/main/webapp/js/classes/primitive-type.js
class PrimitiveType {

    static fromJson(json) {
        if (json == null) {
            return null;
        }
        const primitiveType = new PrimitiveType();

        primitiveType.setId(json.id);
        primitiveType.setName(json.name);
        primitiveType.setIsBaseType(json.isBaseType);
        primitiveType.setIsNumberBaseType(json.isNumberBaseType);
        primitiveType.setIsStreamParamType(json.isStreamParamType);
        primitiveType.setIsArrayType(json.isArrayType);
        primitiveType.setIsRecordType(json.isRecordType);

        return primitiveType;
    }

    constructor() {
        this._id    = null;
        this._name  = null;
        this._isBaseType = false;
        this._isNumberBaseType = false;
        this._isStreamParamType = false;
        this._isArrayType = false;
        this._isRecordType = false;
    };

    setId(id) {
        this._id = id;
    }

    getId() {
        return this._id;
    }

    setName(name) {
        this._name = name;
    }

    getName() {
        return this._name;
    }

    setIsBaseType(isBaseType) {
        this._isBaseType = isBaseType;
    }

    isBaseType() {
        return this._isBaseType;
    }

    setIsNumberBaseType(isNumberBaseType) {
        this._isNumberBaseType = isNumberBaseType;
    }

    isNumberBaseType() {
        return this._isNumberBaseType;
    }

    setIsStreamParamType(isStreamParamType) {
        this._isStreamParamType = isStreamParamType;
    }

    isStreamParamType() {
        return this._isStreamParamType;
    }

    setIsArrayType(isArrayType) {
        this._isArrayType = isArrayType;
    }

    isArrayType() {
        return this._isArrayType;
    }

    setIsRecordType(isRecordType) {
        this._isRecordType = isRecordType;
    }

    isRecordType() {
        return this._isRecordType;
    }
}